content/blog/2021/06/writing-your-first-policy-with-kubewarden.md

```diff
@@ -0,0 +1,553 @@
+book](https://docs.kubewarden.io/writing-policies/index.html#programming-language-requirements).
+
+We are going to write our policy in Go, thus, we will use the TinyGo
+compiler.
+
+In general, starting a new policy is a matter of using the policy
+template we have on GitHub, except for Rust, that has a dedicated
+tool. You can find more information [in the
+book](https://docs.kubewarden.io/).
+
+## The policy
+
+We head to the [`go-policy-template`
+repo](https://github.com/kubewarden/go-policy-template) and use that
+template, creating a repository in one of our GitHub organizations.
+
+### Prerequisites
+
+We clone the repository we just generated from the template locally.
+
+Note: you will need `docker` or `podman` installed for this
+exercise. At the time of writing, the policy template's `Makefile`
+calls to the `docker` CLI, so if you use `podman`, make sure you also
+have a `docker` wrapper or symlink in your `$PATH`. This is so you
+don't have to install TinyGo yourself.
+
+Let's perform a sanity check by running `make`:
+
+```console
+$ rm -rf vendor
+$ make
+```
+
+We remove the `vendor` folder so dependencies will be fetched on every
+build. This is just for the sake of ease of build for the blog post:
+you don't need to do `go mod tidy` and `go mod vendor` afterwards when
+we add more dependencies, so you don't need to have any kind of Go
+toolchain with the right versions on your machine -- a simple `make`
+will build it, a little bit slower than vendored, but still. --
+
+We can see that as a result of the previous command, the sample policy
+that the template contains has been built in a file called
+`policy.wasm`.
+
+```console
+$ file policy.wasm
+policy.wasm: WebAssembly (wasm) binary module version 0x1 (MVP)
+```
+
+Looks about right! Now, let's go ahead and write our policy!
+
+### Policy first go
+
+Let's start by editing the `main.go` file. For the sake blog post
+brevity, I'll keep the code on the `main.go` file, so it's
+straightforward to follow. Also, I'll be skipping some other best
+practices, like TDD or even having unit tests, again for the sake of
+brevity.
+
+```go
+package main
+
+import (
+	wapc "github.com/wapc/wapc-guest-tinygo"
+	kubewarden "github.com/kubewarden/policy-sdk-go"
+)
+
+func main() {
+	wapc.RegisterFunctions(wapc.Functions{
+		"validate": func(payload []byte) ([]byte, error) {
+			return kubewarden.AcceptRequest()
+		},
+		"validate_settings": func(payload []byte) ([]byte, error) {
+			return kubewarden.AcceptSettings()
+		},
+	})
+}
+```
+
+Now we have a program, but we don't have a way to know if it does what
+we want. This instance is pretty trivial, right? It does nothing. It
+just accepts the request as if there was no policy at all. If policy
+ghosting was a thing, this would be it.
+
+On the upside, though, we have a Wasm target object that we can run
+everywhere, provided we have the right Wasm engine tooling on the
+host. How cool is that?
```

Comment:
I really like this take, create something that always accepts a request :+1: :clap: